aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains test case classes based on pytest for AEA skill testing."""
+
+import asyncio
+import inspect
+from pathlib import Path
+from queue import Queue
+from types import SimpleNamespace
+from typing import Optional, Tuple, Type, Union, cast
+
+from aea.context.base import AgentContext
+from aea.identity.base import Identity
+from aea.mail.base import Address
+from aea.multiplexer import AsyncMultiplexer, Multiplexer, OutBox
+from aea.protocols.base import Message
+from aea.skills.base import Skill
+from aea.skills.tasks import TaskManager
+
+
+class BaseSkillTestCase:
+    """A class to test a skill."""
+
+    path_to_skill: Union[Path, str] = Path(".")
+    _skill: Skill
+    _multiplexer: AsyncMultiplexer
+    _outbox: OutBox
+
+    @property
+    def skill(self) -> Skill:
+        """Get the skill."""
+        try:
+            value = self._skill
+        except AttributeError:
+            raise ValueError("Ensure skill is set during setup_class.")
+        return value
+
+    def get_quantity_in_outbox(self) -> int:
+        """Get the quantity of envelopes in the outbox."""
+        return self._multiplexer.out_queue.qsize()
+
+    def get_message_from_outbox(self) -> Optional[Message]:
+        """Get message from outbox."""
+        if self._outbox.empty():
+            return None
+        envelope = self._multiplexer.out_queue.get_nowait()
+        return envelope.message
+
+    @staticmethod
+    def message_has_attributes(
+        actual_message: Message, message_type: Type[Message], **kwargs,
```

Comment:
See if you can infer the type from the performative provided. Then all that's needed is the performative as a required argument, reducing one more line of code

Reply:
I couldn't immediately find a solution for this. I posted a question online, if I get an answer I'll apply it.